File: best-time-to-buy-ii.cc
#include <vector>
using namespace std;

class Solution {
public:
  int maxProfit(vector<int> &prices) {
    if (prices.empty()) return 0;
    int ans = 0;
    int i = 0;
    while (i < prices.size()) {
      int j = i + 1;
      while (j < prices.size() && prices[j] > prices[j - 1]) {
        j++;
      }
      ans += prices[j - 1] - prices[i];
      i = j;
    }
    return ans;
  }
};

File: best-time-to-buy-ii.java
public class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length == 0) return 0;
        int ans = 0;
        for (int i = 0; i < prices.length - 1; i++) {
            if (prices[i] < prices[i + 1])
                ans += prices[i + 1] - prices[i];
        }
        return ans;
    }
}
